### PR TITLE
feat(server): emit RouteDescriptor for raw RouteRegistrar.Add routes

### LINKED ISSUE #634: server: raw RouteRegistrar.Add routes missing from RouteDescriptor registry
## Problem

`server.DefaultRouteRegistry` (`server/descriptor.go`) records a `RouteDescriptor{Method, Path, HandlerID, ...}` for every **typed** route registered through `server.GET/POST/...`, but routes registered through the raw `RouteRegistrar.Add(method, path, Handler)` path never produce a descriptor.

Verified empirically (v0.45.0): after registering one typed and one raw route through the public API, the registry contains only the typed one.

## Why it matters

Consumers building startup inventories keyed by route template (metrics pre-registration, authz matrices, route documentation) silently miss every raw route. This became more visible while diagnosing the `HandlerContext` path-access gap (see the route-template restoration issue): the registry is the natural registration-time complement to per-request `RouteTemplate()`, but it only covers half the surface.

## Proposed fix

Emit a `RouteDescriptor` from the raw `Add` path as well (module attribution via the same registrar context typed routes use). Fields that only make sense for typed routes (request/response model info) stay zero-valued; `HandlerName` can fall back to the function name via reflection or stay empty.

Self-contained, additive; independent of the accessor work.

### PR DESCRIPTION
**Pull request by gaborage**
## Problem

`server.DefaultRouteRegistry` recorded a `RouteDescriptor` for every **typed** route (registered through `server.GET/POST/...`), but routes registered through the raw `RouteRegistrar.Add(method, path, Handler)` path never produced a descriptor. Consumers building startup inventories keyed by route template (metrics pre-registration, authz matrices, route documentation) silently missed every raw route.

Root cause: `DefaultRouteRegistry.Register()` was called from exactly one seam — the generic `RegisterHandler[T,R]` behind the typed helpers. The raw path (`routeGroup.Add`) registered with Echo but never touched the registry.

## Fix

`routeGroup.Add` now records a `RouteDescriptor` alongside the Echo registration. Only the fields derivable at the raw seam are populated (method, full path, handler ID/name, caller package); `RequestType`/`ResponseType`/`InboundJOSE`/`OutboundJOSE` stay `nil` and `ModuleName` stays empty, because a raw handler carries no request/response models (tight scope, per the issue).

Key design points:

- **No double-registration.** The framework's `routeGroup` implements the unexported `addEcho` seam, so typed handlers register through `addEcho` (which emits its own descriptor) and never traverse `Add`. Framework-registered routes are counted exactly once.
- **Single source of truth for `HandlerID`.** Extracted `formatHandlerID(method, fullPath)`, used by both paths, so the `"METHOD:/path"` identifier can't drift — consumers keying inventories by `HandlerID` depend on it.
- **Shared caller attribution.** `getCallerPackage()` → `getCallerPackage(skip int)` so both paths attribute the registering caller's package at their respective stack depths (typed = 3, raw = 2), best-effort, matching existing typed behavior.
- **Nil-field documentation.** `RouteDescriptor` now documents which fields are `nil`/empty for raw routes, so registry consumers nil-check.

## Review handling

`/code-review` (xhigh) + `/security-audit` were run on the diff. Security gates clean (gosec 0 issues, no raw-SQL, no secrets). Of 9 code-review findings:

- **Applied (4):** de-duplicated the `relativePath` recomputation in `Add`; removed a dead `*routeGroup` cast in the new test; documented the nil-for-raw descriptor fields; strengthened the skip-depth assertion.
- **Pushed back (5), with reasoning:** the `getCallerPackage` skip fragility and unbounded-global-accumulation are pre-existing and identical to the typed path; debug endpoints appearing in the registry is the intended feature (raw routes *should* appear); double-registration via a non-`addEcho` wrapper cannot be reached by the framework's own registrar; middleware isn't auto-recorded for typed routes either. No runtime code reads `DefaultRouteRegistry`, so the consumer-gated findings are defensive-only.

The skip-depth test is pinned as far as an in-package test allows; fully validating `skip=2` would require a `package server_test` black-box file, deliberately not added to preserve the repo's 1:1 source-to-test file convention (the limitation is documented in the test).

## Test plan

- New `TestRouteGroupAddEmitsRouteDescriptor` / `TestRouteGroupAddDescriptorUsesFullPath` (RED → GREEN).
- `make check` green: fmt, lint (incl. gosec), race tests across all packages, the ADR-026 alloc-stability guard, and govulncheck.

Closes #634

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Raw route registrations now appear in the route registry with readable method/path details, making routes easier to discover.
  * Handler identifiers are now generated in a consistent `METHOD:/path` format.

* **Bug Fixes**
  * Improved path handling so nested routes are recorded using their full resolved path.
  * Clarified behavior for raw routes so metadata is handled more reliably when some fields are unavailable.

### CODE REVIEW

File: server/route_registrar.go [hunk 1]
```diff
@@ -31,8 +31,25 @@ func (rg *routeGroup) addEcho(method, path string, h echo.HandlerFunc) {
...
+
+	fullPath := rg.fullPathFromRelative(relative)
+	DefaultRouteRegistry.Register(&RouteDescriptor{
+		Method:      method,
+		Path:        fullPath,
+		HandlerID:   formatHandlerID(method, fullPath),
+		HandlerName: extractHandlerName(handler),
+		Package:     getCallerPackage(2), // getCallerPackage → Add → module (best-effort, as typed routes)
+	})
 }
```
[the remark this answers is not in the document]
Comment (gaborage) [reply server/route_registrar.go:53]: Intentionally skipped — the `Middleware` field is not a protection oracle, and populating it here would give false confidence rather than fix one.

Verified against the current tree:

- `RouteDescriptor.Middleware` is set **only** via the opt-in `WithMiddleware(names ...string)` route option (`descriptor.go:129`, `:143`). It is *never* auto-derived from actually-applied middleware — not for typed routes either. `WithMiddleware(` has **zero call sites** in the framework, and nothing reads `descriptor.Middleware` at runtime (`DefaultRouteRegistry` is in-process introspection only; the OpenAPI generator is static-analysis in a separate repo).
- So "empty `Middleware` ⇒ unprotected" is already false for the entire registry today: typed routes carry group-level guards (`Group(...)`/`Use(...)`) and echo-native middleware that the field has never reflected. Raw routes are consistent with that, not newly inconsistent.
- Capturing only the per-route `...MiddlewareFunc` passed to `Add` would **omit group-level middleware**, so an authz tool would still see an incomplete guard set — but now with the false confidence of a populated field. It would also reflect anonymous middleware closures as `func1`/`func2` noise and mix reflected names into a field whose only defined semantic is curated human-supplied strings.
- The middleware **is** applied to echo (`rg.adaptAll(middleware)` at `route_registrar.go`) — enforcement is unaffected. This is purely about descriptor metadata.

This stays out of the deliberately tight, additive scope of #634 (method/path/handler/package attribution, matching the typed path). If descriptor-based middleware inventory is wanted later, the right design is a single seam that captures group- and route-level middleware for *both* typed and raw routes — a separate, larger change tracked on its own.
